pkg/util/paramtable/component_param.go

```diff
@@ -1439,6 +1440,25 @@ func (p *queryNodeConfig) init(base *BaseTable) {
 	}
 	p.StatsPublishInterval.Init(base.mgr)
 
+	p.KnowhereThreadPoolSize = ParamItem{
+		Key:          "queryNode.segcore.knowhereThreadPoolNumRatio",
+		Version:      "2.0.0",
+		DefaultValue: "4",
+		Formatter: func(v string) string {
+			factor := getAsInt64(v)
+			if factor <= 0 || !p.EnableDisk.GetAsBool() {
+				factor = 1
+			} else if factor > 16 {
+				factor = 16
+			}
+			knowhereThreadPoolSize := uint32(runtime.GOMAXPROCS(0)) * uint32(factor)
+			return strconv.FormatUint(uint64(knowhereThreadPoolSize), 10)
+		},
+		Doc:    "The number of threads in knowhere's thread pool. If disk is enabled, the pool size will increase by the knowhereThreadPoolNumRatio.",
```

Comment:
Doc needs to write about how to handle values less than 0 and greater than 16.
It should sync with comment in milvus.yaml